### ISSUE TITLE
jdeps can't find javafx modules on Java 20 toolchain

### ISSUE THREAD

**Issue by lenis0012** [hydraulic-software/conveyor#55]
**Describe the bug**
jdeps can't find javafx modules for OpenJFX 20 on Java 20, resulting in fallback to jar-based deployment and an unoptimized JRE.

Error when running `conveyor -v make windows-app --rerun`:
```
❌ jdeps failed scanning JAR oraksi-1.0.jar
❯ [Processed Jars for Windows Intel] jdeps failed scanning JAR oraksi-1.0.jar
hydraulic.shell.SubprocessFailed: C:\Users\Lennart\AppData\Local\Packages\Conveyor_fg3qp2cw01ypp\LocalCache\Local\Hydraulic\Conveyor\Cache\entries\9c\00\9c0091819b86a1dc\content/bin/jdeps: Returned non-zero exit code 1
Exception in thread "main" java.lang.module.FindException: Module javafx.controls not found, required by com.lenis0012.oraksi
        at java.base/java.lang.module.Resolver.findFail(Resolver.java:892)
        at java.base/java.lang.module.Resolver.resolve(Resolver.java:192)
        at java.base/java.lang.module.Resolver.resolve(Resolver.java:141)
        at java.base/java.lang.module.Configuration.resolve(Configuration.java:420)
        at java.base/java.lang.module.Configuration.resolve(Configuration.java:254)
        at jdk.jdeps/com.sun.tools.jdeps.JdepsConfiguration$Builder.build(JdepsConfiguration.java:564)
        at jdk.jdeps/com.sun.tools.jdeps.JdepsTask.buildConfig(JdepsTask.java:604)
        at jdk.jdeps/com.sun.tools.jdeps.JdepsTask.run(JdepsTask.java:558)
        at jdk.jdeps/com.sun.tools.jdeps.JdepsTask.run(JdepsTask.java:534)
        at jdk.jdeps/com.sun.tools.jdeps.Main.main(Main.java:50) @ hydraulic.shell.impl.ExecKt.waitForSubProcess(Unknown Source)
```

**To Reproduce**
1. Generate new javafx application using conveyor
2. Make sure JDK 20 is installed and active
3. Set toolchain version and jfx version to `20` in `build.gradle`
4. Set gradle wrapper version to `8.1-rc-1` in `gradle/wrapper/gradle-wrapper.properties` to support JDK 20
5. Run `conveyor -v make windows-app`
6. See error

**Expected behavior**
No errors, optimized JRE using module dependency graph and jimage deployment

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 11

**Additional context**
Not 100% sure if this is related to Java 20 or JFX 20 but this issue did not occur when using the default (JFX 19 & java 17)

**Comment by lenis0012** [comment]
Changing JDK also causes errors because conveyor caches the jdeps binary.

Perhaps it is a better idea to scan the dependency graph from within gradle? there is a gradle jdeps plugin that can be used as dependency or for inspitation.

That way the classpath and toolchain are managed by gradle.

**Comment by lenis0012** [comment]
I can confirm that the issue is unrelated to the gradle or jfx update and only happens when you update the gradle toolchain to 20.

**Comment by mikehearn** [comment]
Could you attach a log please? We can't reproduce this. I was able to upgrade both Gradle and OpenJDK to 20 and the build completed successfully. The jdeps error is talking about a third party library, not jfx itself. Maybe there's something odd about that jar that the new jdeps rejects and old jdeps is OK with?

**Comment by lenis0012** [comment]
[log.24.zip](https://github.com/hydraulic-software/conveyor/files/11137422/log.24.zip)

**Comment by lenis0012** [comment]
System path:
![Screenshot 2023-04-01 222704](https://user-images.githubusercontent.com/1535717/229488079-bb3028f4-0765-4cb6-b4eb-653f43859c8a.png)

I made a new projecr with only the changes I described and no other modifications and it still occurs:

❌ jdeps failed scanning JAR jdeps-test-1.0.jar

**Comment by lenis0012** [comment]
I see. thank you and I apologize for misunderstanding.

**Comment by lenis0012** [comment]
It was my understanding from the documentaiton that your application is supposed to be packaged into the modules file and I thought that the jdeps error was preventing that from happening:

from the docs:
```

**Comment by mikehearn** [comment]
It's not your mis-understanding, there's a real bug, it's just not especially serious if you already know your dependencies.

The only jars that can go `modules` are jars that are both explicit and which don't depend on auto modules. That's a jlink limitation, not ours. Conveyor scans the jars to figure out what that set is so they can go into the modules file. It doesn't work if it encounters a module with a java 20 `module-info.class` file and won't add it into the `modules` file in that case.

It's arguable what is more desirable. On one hand, being jlinked in to `images` means slightly better startup time (in theory), and slightly smaller downloads. On the other hand it reduces the effectiveness of delta updates.

**Comment by lenis0012** [comment]
conveyor supports delta updates? I didn't know that

**Comment by mikehearn** [comment]
On Windows only. On macOS the update engine we ship (sparkle) also supports delta updates but Conveyor doesn't currently make them as part of the build process.

**Comment by mikehearn** [comment]
We shipped a fix for the jdeps issue in 8.1, please upgrade and see if that works for you.
